### DISCUSSION TITLE
Exception handling in steps with tables

### DISCUSSION THREAD

**Discussion by ademant** [getgauge/gauge#2199] [General]
According to your description (https://docs.gauge.org/writing-specifications.html?os=macos&language=javascript&ide=vscode#step-implementations) I wrote following step definition to check all input fields on a site:
```
step("Check field and value via table <table>",
  async function (table) {
      table.entries(function(entry) {
        checkName(driver,entry["value"],entry["field"]);
      });
  }
);

```
In a scenario I like to use it as followed:
```
* Check field and value via table
| field | value |
|----|----|
| name | test |
| description | Description |

```
If values differ from expected value the step does not give an error, but gives a green check. I tried several tricks, changing to sync functions, use try/catch statements, used .catch functions, but none worked. Differing values are accepted with the test.

Environment:
Ubuntu Impish
Gauge 1.4.3
html-report 4.1.4
js 2.4.0
json-report 0.3.6

**Comment by zabil** [comment]
Thanks for pointing that out. This is an [issue](https://github.com/getgauge/gauge/issues/2200). You can use a workaround of using rows instead for example.

https://github.com/getgauge/template-js-simple/blob/1b53b83c676d0f00b79cddaeff5de4426f12b64b/tests/step_implementation.js#L26-L30

**Reply by ademant** [reply]
Thank you for your answer. I tried the workaround, but it didn't work well.
In my case I have to work directly with a web page and use chromedriver. For this I use for example code like:
`
let element = webdriver.findElement(By.name(name));
let content = element.getAttribute("value");
assert.equal(content,toCompare);
`
Sometimes the async code lead to a condition that the result of a promised check comes to late.
